assingments/linsifa/quicksort1.py

```diff
@@ -0,0 +1,78 @@
+arr = [54, 7, 25, 2, 6, 92, 8, 12, 13]
+
+# values.sort()
+
+# print("sorted values: ", values)
+def partition(arr,low,high):
+
+    i = ( low-1 )         # index of smaller element
+
+    pivot = arr[high]     # pivot
+
+  
+
+    for j in range(low , high):
+
+  
+
+        # If current element is smaller than or
+
+        # equal to pivot
+
+        if   arr[j] < pivot:
+
+          
+
+            # increment index of smaller element
+
+            i = i+1
+
+            arr[i],arr[j] = arr[j],arr[i]
+
+  
+
+    arr[i+1],arr[high] = arr[high],arr[i+1]
+
+    return ( i+1 )
+
+  
+
+# The main function that implements QuickSort
+
+# arr[] --> Array to be sorted,
+
+# low  --> Starting index,
+
+# high  --> Ending index
+
+  
+
+# Function to do Quick sort
+
+def quickSort(arr,low,high):
+
+    if low < high:
+
+  
+
+        # pi is partitioning index, arr[p] is now
+
+        # at right place
+
+        pi = partition(arr,low,high)
+
+  
+
+        # Separately sort elements before
+
+        # partition and after partition
+
+        quickSort(arr, low, pi-1)
+
+        quickSort(arr, pi+1, high)
+
+       
+
+quickSort(arr, 0, len(arr)-1)
+
+print("sorted values: ", arr)
```

Comment:
remove print statement and add return for non inplace functons. 
No print statements.

Reply:
rename the file to quicksort.py not quicksort1.py

Reply:
also rebase. check video.